app/infra/message_senders/base.py

```diff
@@ -0,0 +1,9 @@
+from abc import abstractmethod, ABC
+from dataclasses import dataclass
+
+
+@dataclass
+class BaseMessageSender(ABC):
+    @abstractmethod
+    def send(self, **kwargs) -> None:
+        raise NotImplementedError()
```

Comment:
You don't need to raise an exception here - it is done automatically then you have @abstractmethod property. It's better to substitute it with pass